device/alpaka/include/traccc/alpaka/utils/queue.hpp

```diff
@@ -0,0 +1,63 @@
+/**
+ * traccc library, part of the ACTS project (R&D line)
+ *
+ * (c) 2025 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// System include(s).
+#include <cstddef>
+#include <limits>
+#include <memory>
+
+namespace traccc::alpaka {
+
+/// Owning wrapper around @c ::alpaka::Queue
+class queue {
+
+    public:
+    /// Invalid/default device identifier
+    static constexpr std::size_t INVALID_DEVICE =
+        std::numeric_limits<std::size_t>::max();
+
+    /// Construct a new stream (possibly for a specified device)
+    explicit queue(std::size_t device = INVALID_DEVICE);
+
+    /// Wrap an existing ::alpaka::Queue object
+    ///
+    /// Without taking ownership of it!
+    ///
+    queue(void* input_queue);
```

Comment:
To be **very** safe, you could add `explicit` to this constructor. Just as how SonarCloud recommends.